### PR TITLE
Add simple cpp demo

### PR DESCRIPTION
**Pull request by IgorBaratta**
Shows how to link basix (cmake),  and useful for testing in hpc enviroment.

### CODE REVIEW

**Review by mscroggs — APPROVED** [review]
looks good

File: .github/workflows/dolfin-tests.yml [hunk 1]
```diff
@@ -39,6 +39,13 @@ jobs:
           python3 -m pip install git+https://github.com/FEniCS/ufl.git
           python3 -m pip install git+https://github.com/FEniCS/ffcx.git
 
+      - name: Run cpp demos
```
Comment (mscroggs) [inline .github/workflows/dolfin-tests.yml:42]: Is this intentionally in the DOLFINx tests workflow? Would it make more sense in pythonapp.yml?

**Comment by mscroggs** [comment]
I get `/usr/local/lib/libbasix.so: undefined reference to `dladdr'` when I try to make the demo locally

File: .github/workflows/dolfin-tests.yml (hunk 1 above)
Comment (IgorBaratta) [reply .github/workflows/dolfin-tests.yml:42]: pythonapp.yml installs basix with pip, so it's not installed in the default foder.

**Comment by IgorBaratta** [comment]
> I get `/usr/local/lib/libbasix.so: undefined reference to `dladdr'` when I try to make the demo locally

I've updated the CMakeLists.txt, could you check if it fixes the issue?

**Review by mscroggs — APPROVED** [review]

**Pull request merged**